monai/transforms/utils.py

```diff
@@ -837,3 +843,90 @@ def compute_divisible_spatial_size(spatial_shape: Sequence[int], k: Union[Sequen
         new_size.append(new_dim)
 
     return new_size
+
+
+def convert_to_tensor(data):
+    """
+    Utility to convert the input data to a PyTorch Tensor. If passing a dictionary, list or tuple,
+    recursively check every item and convert it to PyTorch Tensor.
+
+    Args:
+        data: input data can be PyTorch Tensor, numpy array, list, dictionary, int, float, bool, str, etc.
+            will convert Tensor, Numpy array, float, int, bool to Tensors, strings and objects keep the original.
+            for dictionay, list or tuple, convert every item to a Tensor if applicable.
+
+    """
+    if isinstance(data, torch.Tensor):
+        return data.contiguous()
+    elif isinstance(data, np.ndarray):
+        # skip array of string classes and object, refer to:
+        # https://github.com/pytorch/pytorch/blob/v1.9.0/torch/utils/data/_utils/collate.py#L13
+        if re.search(r"[SaUO]", data.dtype.str) is None:
+            # numpy array with 0 dims is also sequence iterable,
+            # `ascontiguousarray` will add 1 dim if img has no dim, so we only apply on data with dims
+            return torch.as_tensor(data if data.ndim == 0 else np.ascontiguousarray(data))
+    elif isinstance(data, (float, int, bool)):
+        return torch.as_tensor(data)
+    elif isinstance(data, dict):
+        return {k: convert_to_tensor(v) for k, v in data.items()}
+    elif isinstance(data, list):
+        return [convert_to_tensor(i) for i in data]
+    elif isinstance(data, tuple):
+        return tuple([convert_to_tensor(i) for i in data])
+
+    return data
```

Comment:
I think this should raise an error, otherwise it's not 'ensure'
```suggestion
    raise ValueError
```

Reply:
Hi @wyli ,

Thanks for your review.
This `return data` is to support recursive operation for the `string` or objects in a dict or list. For example, in `meta dict` we can convert the meta_dict: `meta_dict = {"img": np.array([1, 2, 3]), "filename_or_obj": "./workspace/data/img.nii"}`.
What do you think?

Thanks.

Reply:
Ok, perhaps we should clarify this in the ensuretype transform. This utility looks fine to me.

Reply:
Sure, enhanced the doc-string the latest commit.
Thanks.